onnxruntime/core/providers/shared_library/provider_bridge_provider.cc

```diff
@@ -291,17 +291,11 @@ std::vector<std::unique_ptr<ComputeCapability>> IExecutionProvider::GetCapabilit
                                                                                   const std::vector<const KernelRegistry*>& kernel_registries) const {
   return g_host->IExecutionProvider__GetCapability(this, graph_viewer, kernel_registries);
 }
-
+// !!! This API will be deprecated soon.
 common::Status IExecutionProvider::Compile(const std::vector<onnxruntime::Node*>& fused_nodes,
```

Comment:
Could we have a std::call_once that prints a warning on the first usage of the API? Runtime not compile time, but at least an explicit warning

Reply:
Isn't some output better than nothing? Low cost and at least someone gets a hint that an update is required.

Reply:
add a std::call_once warning message when the legacy compile first been called.